staging/src/k8s.io/kube-openapi/pkg/aggregator/aggregator.go

```diff
@@ -0,0 +1,480 @@
+	}
+	if !equalStringArray(s1.Type, s2.Type) {
+		return false
+	}
+	if s1.Format != s2.Format {
+		return false
+	}
+	if !equalFloatPointer(s1.Minimum, s2.Minimum) {
+		return false
+	}
+	if !equalFloatPointer(s1.Maximum, s2.Maximum) {
+		return false
+	}
+	if s1.ExclusiveMaximum != s2.ExclusiveMaximum {
+		return false
+	}
+	if s1.ExclusiveMinimum != s2.ExclusiveMinimum {
+		return false
+	}
+	if !equalFloatPointer(s1.MultipleOf, s2.MultipleOf) {
+		return false
+	}
+	if !equalIntPointer(s1.MaxLength, s2.MaxLength) {
+		return false
+	}
+	if !equalIntPointer(s1.MinLength, s2.MinLength) {
+		return false
+	}
+	if !equalIntPointer(s1.MaxItems, s2.MaxItems) {
+		return false
+	}
+	if !equalIntPointer(s1.MinItems, s2.MinItems) {
+		return false
+	}
+	if s1.Pattern != s2.Pattern {
+		return false
+	}
+	if s1.UniqueItems != s2.UniqueItems {
+		return false
+	}
+	if !equalIntPointer(s1.MaxProperties, s2.MaxProperties) {
+		return false
+	}
+	if !equalIntPointer(s1.MinProperties, s2.MinProperties) {
+		return false
+	}
+	if !equalStringArray(s1.Required, s2.Required) {
+		return false
+	}
+	return len(s1.Enum) == 0 && len(s2.Enum) == 0 && len(s1.Dependencies) == 0 && len(s2.Dependencies) == 0
+}
+
+func renameDefinition(s *spec.Swagger, old, new string) {
+	old_ref := DefinitionPrefix + old
+	new_ref := DefinitionPrefix + new
+	walkOnAllReferences(func(ref spec.Ref) spec.Ref {
+		if ref.String() == old_ref {
+			return spec.MustCreateRef(new_ref)
+		}
+		return ref
+	}, s)
+	s.Definitions[new] = s.Definitions[old]
+	delete(s.Definitions, old)
+}
+
+// Copy paths and definitions from source to dest, rename definitions if needed.
+// dest will be mutated, and source will not be changed.
+func MergeSpecs(dest, source *spec.Swagger) error {
+	source, err := CloneSpec(source)
+	if err != nil {
+		return err
+	}
+	for k, v := range source.Paths.Paths {
+		if _, found := dest.Paths.Paths[k]; found {
+			return fmt.Errorf("Unable to merge: Duplicated path %s", k)
+		}
+		dest.Paths.Paths[k] = v
+	}
+	usedNames := map[string]bool{}
+	for k := range dest.Definitions {
+		usedNames[k] = true
+	}
+	type Rename struct {
+		from, to string
+	}
+	renames := []Rename{}
+	for k, v := range source.Definitions {
+		v2, found := dest.Definitions[k]
+		if found || usedNames[k] {
+			if found && EqualSchema(&v, &v2) {
+				continue
+			}
+			i := 2
+			newName := fmt.Sprintf("%s_v%d", k, i)
+			for usedNames[newName] {
+				i += 1
+				newName = fmt.Sprintf("%s_v%d", k, i)
+			}
+			renames = append(renames, Rename{from: k, to: newName})
+			usedNames[newName] = true
+		} else {
+			usedNames[k] = true
+		}
+	}
+	for _, r := range renames {
+		renameDefinition(source, r.from, r.to)
+	}
+	for k, v := range source.Definitions {
+		if _, found := dest.Definitions[k]; !found {
+			dest.Definitions[k] = v
+		}
+	}
+	return nil
+}
+
+// Clone OpenAPI spec
+func CloneSpec(source *spec.Swagger) (*spec.Swagger, error) {
+	if ret, err := cloner.DeepCopy(source); err != nil {
```

Comment:
json marshaling is fine. I am doing the same in my pending static deepcopy PR, for this very code line.